frontend/public/components/configmap-and-secret-data.tsx

```diff
@@ -76,20 +76,18 @@ export const SecretValue: React.FC<SecretValueProps> = ({ value, reveal, encoded
 };
 SecretValue.displayName = 'SecretValue';
 
-export const SecretData: React.FC<SecretDataProps> = ({ data, title = 'Data' }) => {
+export const SecretData: React.FC<SecretDataProps> = ({ data, title = 'Data', sort = true }) => {
   const [reveal, setReveal] = React.useState(false);
 
   const dl = [];
-  Object.keys(data || {})
-    .sort()
-    .forEach((k) => {
-      dl.push(<dt key={`${k}-k`}>{k}</dt>);
-      dl.push(
-        <dd key={`${k}-v`}>
-          <SecretValue value={data[k]} reveal={reveal} />
-        </dd>,
-      );
-    });
+  (sort ? Object.keys(data || {}).sort() : Object.keys(data || {})).forEach((k) => {
```

Comment:
`data` is a map, so it has no guaranteed order.

Reply:
```
type KeyValueData = {
  [key: string]: string;
};
```
I see that `data` is of type `KeyValueData`. Essentially an object. Doesn't object retain insertion order as of ES2015?

Reply:
That's JS, not JSON. The API itself guarantees no order.

Reply:
If you see my use case, I am using an Object to maintain order and pass it to SecretData. I am not directly passing JSON data.

Reply:
Oh, we shouldn't be using this component if it's not a real secret IMO. That's confusing.

Much better to use an array internally as well if order matters. Key insertion order sort-of works, but there are some edge cases that break (for instance, keys like "1" don't maintain key insertion order in Chrome).

Reply:
Not using the secret data component also avoids the artificial `Base64.encode`.

Reply:
What do you suggest? Write a similar component to `SecretData`? I think it's possible for me to reuse SecretValue(and other components) and all the CSS involved there. But I feel that would be repeating almost the same component. I am not using an integer as indices over here though but I get the point.

Reply:
It should be a very small amount of code, just a `dl` that uses `SecretValue` directly for the values. We've made `SecretValue` general, so it's reusable outside of this component and already use it elsewhere. You can pass `encoded={false}` to skip the artificial encoding. This is much cleaner imo.

Reply:
will do. thanks for the prompt review. :)